chain/chain/src/store.rs

```diff
@@ -1919,6 +1934,13 @@ impl<'a> ChainStoreUpdate<'a> {
                         .map(|trie_changes: TrieChanges| {
                             tries
                                 .revert_insertions(&trie_changes, shard_id, &mut store_update)
+                                .map(|_| {
+                                    ColTrieChanges.gc(
+                                        &get_block_shard_id(&block_hash, shard_id).into(),
+                                        &mut store_update,
+                                        GC_NOCACHE,
+                                    )
+                                })
```

Comment:
Awesome catch.

Reply:
Should gc on ColTrieChanges handle the arounded logic, similar to ColBlockPerHeight?

Reply:
It should be inside `gc_col`.